### DISCUSSION TITLE
How to save large files using BackgroundTasks or equivalent?

### DISCUSSION THREAD

**Discussion by ArturFortunato** [fastapi/fastapi#11817] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@staticmethod
def upload_video(
    background_tasks: BackgroundTasks,
    video: UploadFile = File(...),
):
    background_tasks.add_task(ProjectService.upload_video, video)
    return {}
```

### Description

My endpoints receives a very large video (2h with 1080p resolution). My function `ProjectService.upload_video` saves that video into a file. If I try to move `ProjectService.upload_video(video)` to `backgroud_tasks`, I get

```
Traceback (most recent call last):
  File "/app/./app/services/entities/Project.py", line 34, in upload_video
    video_location = upload_video_service.upload_video(video, project_id)
  File "/app/./app/services/Video.py", line 74, in upload_video
    file.write(video.file.read())
  File "/usr/local/lib/python3.10/tempfile.py", line 743, in read
    return self._file.read(*args)
ValueError: read of closed file
```

I found a similar request [here](https://github.com/tiangolo/fastapi/discussions/6108), but there was no explicit answer on how to actually use BackgroundTasks for this.

So questions are:
1. How can prevent the file description from being closed
2. Is there a better way to force it directly into a file path that I choose (or get the path of the temporary file the FastApi creates? (This would be awesome!)

Thanks!

### Operating System

Linux

### FastAPI Version

0.109.2

### Pydantic Version

1.10.2

### Python Version

3.9.6

Edit: related thread [here](https://github.com/tiangolo/fastapi/issues/10857), possibly related to a breaking change release on 0.106.0

**Comment by ArturFortunato** [comment]
Can confirm that downgrading to 0.105.0 fixed the issue

**Reply by YuriiMotov** [reply]
That's because in 0.106.0 the braking change was introduced. Since then the background tasks are executed after the response is sent and browser can close the connection. So, at this point you can not longer read the request.

Can you describe why you need to move it to background tasks?

**Comment by 4c0n** [comment, marked as answer]
I don't think it makes sense what you seem to be trying to do. As long as data is coming in, you would want to stay in the path operation context, as that is where the data is coming in. You could however read the input data in chunks and write the chunks to save your precious memory. At least I suspect that is what you're trying to achieve?
